src/core/ensemble_scheduler.cc

```diff
@@ -32,21 +32,64 @@
 #include "src/core/backend.h"
 #include "src/core/cuda_utils.h"
 #include "src/core/logging.h"
+#include "src/core/metrics.h"
 #include "src/core/server.h"
 
 namespace nvidia { namespace inferenceserver {
 
 namespace {
 
-// Step specifies the backend, providers and status objects used for
-// the internal infer request
+DataType
```

Comment:
This and the inverse is available in tritonserver.cc. Should move them to model_config.cc so they can be shared.